src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -81,47 +110,45 @@ public static String printReportMarkdown(PullRequestRef pr, String stashURL, Str
       
       sb.append("| Total New Issues | ").append(issueNumber).append(" |").append(NEW_LINE);
       sb.append("|-----------------|------|").append(NEW_LINE);
-      sb.append(printIssueNumberBySeverityMarkdown(report, coverageReport, Severity.BLOCKER));
-      sb.append(printIssueNumberBySeverityMarkdown(report, coverageReport, Severity.CRITICAL));
-      sb.append(printIssueNumberBySeverityMarkdown(report, coverageReport, Severity.MAJOR));
-      sb.append(printIssueNumberBySeverityMarkdown(report, coverageReport, Severity.MINOR));
-      sb.append(printIssueNumberBySeverityMarkdown(report, coverageReport, Severity.INFO));
+      for (String severity: orderedSeverities) {
+        sb.append(printIssueNumberBySeverityMarkdown(allIssues, severity));
+      }
       sb.append(NEW_LINE).append(NEW_LINE);
   
       // Issue list
       sb.append("| Issues list |").append(NEW_LINE);
       sb.append("|------------|").append(NEW_LINE);
-      sb.append(printIssueListBySeverityMarkdown(report, sonarQubeURL, Severity.BLOCKER));
-      sb.append(printIssueListBySeverityMarkdown(report, sonarQubeURL, Severity.CRITICAL));
-      sb.append(printIssueListBySeverityMarkdown(report, sonarQubeURL, Severity.MAJOR));
-      sb.append(printIssueListBySeverityMarkdown(report, sonarQubeURL, Severity.MINOR));
-      sb.append(printIssueListBySeverityMarkdown(report, sonarQubeURL, Severity.INFO));
+      for (String severity: orderedSeverities) {
+        sb.append(printIssueListBySeverityMarkdown(generalIssues, sonarQubeURL, severity));
+      }
       sb.append(NEW_LINE).append(NEW_LINE);
     }
     
     // Code coverage
-    if (! coverageReport.isEmpty()) {
-      sb.append(printCoverageReportMarkdown(stashProject, stashRepo, pullRequestId, coverageReport, stashURL));
+    if (!coverageIssues.isEmpty()) {
+      sb.append(printCoverageReportMarkdown(stashProject, stashRepo, pullRequestId, coverageIssues, stashURL, projectCoverage, previousProjectCoverage, issuePathResolver));
     }
 
     return sb.toString();
   }
   
-  public static String printCoverageReportMarkdown(String stashProject, String stashRepo, int pullRequestId, CoverageIssuesReport coverageReport, String stashURL) {
-    StringBuilder sb = new StringBuilder("| Code Coverage: ");
+  public static String printCoverageReportMarkdown(String stashProject, String stashRepo, int pullRequestId, List<Issue> coverageReport, String stashURL,
+                                                   Double projectCoverage, Double previousProjectCoverage, IssuePathResolver issuePathResolver) {
+    StringBuilder sb = new StringBuilder("| Line Coverage: ");

```

Comment:
This would be a future feature after this is done.
Can you open a feature request?